addons/uom/models/uom_uom.py

```diff
@@ -13,11 +13,60 @@ class UoMCategory(models.Model):
 
     name = fields.Char('Unit of Measure Category', required=True, translate=True)
 
+    uom_ids = fields.One2many('uom.uom', 'category_id')
+    reference_uom_id = fields.Many2one('uom.uom', "Reference UoM", store=False, help="Dummy field to keep track of reference uom change")
+
+    @api.model
+    def create(self, values):
+        rec = super(UoMCategory, self.with_context(no_unique_reference_check=True)).create(values)
+        if len(rec.uom_ids) > 1:
+            rec._check_category_reference_uniqueness()
+        return rec
+
+    def write(self, values):
+        super(UoMCategory, self.with_context(no_unique_reference_check=True)).write(values)
+        if len(self.uom_ids) > 1:
+            self._check_category_reference_uniqueness()
+
+    def _check_category_reference_uniqueness(self):
```

Comment:
You can add the decoration `@api.constrains('uom_ids')` to avoid the override `create` and `write` of this class.
Also you should batch this method. (and then you can remove `for rec in res:` in create of `uom.uom` model).